Model/Queue/Processing.php

```diff
@@ -373,119 +393,97 @@ protected function processWithAvaTax(Queue $queue, $entity)
     /**
      * @param \Magento\Sales\Api\Data\InvoiceInterface|\Magento\Sales\Api\Data\CreditmemoInterface $entity
      * @param \ClassyLlama\AvaTax\Api\Data\GetTaxResponseInterface $processSalesResponse
+     * @throws \Exception
      */
-    protected function updateAdditionalEntityAttributes(
+    protected function saveAvaTaxRecord(
         $entity,
         GetTaxResponseInterface $processSalesResponse
-    ) {
-        $entityExtension = $entity->getExtensionAttributes();
-        if ($entityExtension == null) {
-            $entityExtension = $this->getEntityExtensionInterface($entity);
-        }
+    )
+    {
+        $avaTaxRecord = $this->createAvataxEntity($entity);
+        // Load existing AvaTax entry for this entity, if exists
+        $avaTaxRecord->loadByParentId($entity->getId());
 
-        // check to see if the AvataxIsUnbalanced is already set on this entity
-        $avataxIsUnbalancedToSave = false;
-        if ($entityExtension->getAvataxIsUnbalanced() === null) {
-            $entityExtension->setAvataxIsUnbalanced($processSalesResponse->getIsUnbalanced());
-            $avataxIsUnbalancedToSave = true;
-        } else {
-            // check to see if any existing value is different from the new value
-            if ($processSalesResponse->getIsUnbalanced() <> $entityExtension->getAvataxIsUnbalanced()) {
-                // Log the warning
-                $this->avaTaxLogger->warning(
-                    __('When processing an entity in the queue there was an existing AvataxIsUnbalanced and ' .
-                    'the new value was different than the old one. The old value was overwritten.'),
-                    [ /* context */
-                        'old_is_unbalanced' => $entityExtension->getAvataxIsUnbalanced(),
-                        'new_is_unbalanced' => $processSalesResponse->getIsUnbalanced(),
-                    ]
-                );
-                $entityExtension->setAvataxIsUnbalanced($processSalesResponse->getIsUnbalanced());
+        if ($avaTaxRecord->getData('parent_id')) {
```

Comment:
If the magic getters are specified in the docblock on the class you could call this as `$avaTaxRecord->getParentId()`